validator/sawtooth_validator/execution/processor_handlers.py

```diff
@@ -53,19 +67,37 @@ def handle(self, connection_id, message_content):
             list(request.namespaces),
             max_occupancy)
 
+        # Reject the request if requested version cannot be handled,
+        # validator does backward compatible support
+        if request.protocol_version > SDK_PROTOCOL_VERSION:
+            ack = processor_pb2.TpRegisterResponse()
+            ack.status = ack.ERROR
+            # Send protocol_version of validator, so that SDK can cross verify
+            # if it can get all services requested
+            ack.protocol_version = SDK_PROTOCOL_VERSION
```

Comment:
This should be set to request.protocol_version (the one requested). Same below.

Reply:
Intent was to let transaction processor know valdiator's version in response. Since validator promises backward compatibility transaction processor can decide how it wants to handle the response message from validator. Please let me know if you meant something else in earlier comment.

Reply:
The later protocol versions may introduce incompatibilities (i.e. send things to the TPs that weren't expected in earlier versions). However, the guarantee we are making here is that the validator will operate at exactly the requested protocol version (whether it could have supported later ones or not). Thus the response should match the request. This could be implied, except current validators don't know how to set it, and we are essentially just using this as a mechanism for the SDKs to detect it.

Reply:
Ok, I guess you intend to say processor SDK will always ask for specific version feature. Validator should know upto what maximum version it can support and handle the request as per SDK version. Thus, it's required to send same version in both request and response to the SDK.